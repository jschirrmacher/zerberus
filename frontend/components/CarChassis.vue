<script setup lang="ts">
import ABox from "./ABox.vue"

defineProps<{
  length: number
  width: number
  height: number
  color: string
}>()
</script>

<template>
  <ABox class="car-chassis" :length="length + 'px'" :width="width + 'px'" :height="height + 'px'" :color="color" />
</template>

<style scoped lang="scss"></style>
