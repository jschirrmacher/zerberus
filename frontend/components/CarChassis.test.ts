import { describe, it, expect } from "vitest"
import { shallowMount } from "@vue/test-utils"
import CarChassis from "./CarChassis.vue"

describe("CarChassis", () => {
  it("should render as expected", () => {
    const component = shallowMount(CarChassis, {
      props: { length: 100, width: 200, height: 50, color: "#abcdef" },
    })
    expect(component.element).toMatchSnapshot()
  })
})
