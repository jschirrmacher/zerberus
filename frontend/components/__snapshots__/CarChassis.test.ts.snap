// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarChassis > should render as expected 1`] = `
<a-box-stub
  class="car-chassis"
  color="#abcdef"
  height="50px"
  length="100px"
  width="200px"
/>
`;
